### ISSUE TITLE
Some questions about the architecture

### ISSUE THREAD

**Issue by baptistepouthier** [mpc001/end-to-end-lipreading#12]
Hi! Thank you for your work!
I'm trying to reproduce your network architecture using keras. I'm not sure to understand everything, so i have few questions:

- What is the shape of the tensor / array containing the audio waveforms given to the audio network? Is it (number of samples, waveform's vector size)? If possible, can you print the summary of the model to show the outputs of each layer? It would help me a lot!
- When you tried to use MFCC, did you put something between the MFCC feature and the BGRU layers ? Or did you feed the feature to the BGRU layer directly?
- Did you tried to truncate the LRW videos to isolate the words?

Thank you!

**Comment by mpc001** [comment]
Hi, @baptistepouthier,
Sorry for the late reply!

"What is the shape of the tensor / array containing the audio waveforms given to the audio network? Is it (number of samples, waveform's vector size)? If possible, can you print the summary of the model to show the outputs of each layer? It would help me a lot!"
Of course, please check the outputs of each layer:

```
    def forward(self, x):
        x = x.view(-1, 1, x.size(1))  # input size: (B, 19456); output size: (B, 1, 19456)
        x = self.fronted1D(x)  # output size: (B, 64, 4864)
        x = x.contiguous()
        x = self.resnet18(x)  # output size: (B*29, 512)
        if self.mode == 'temporalConv':
            x = x.view(-1, self.frameLen, self.inputDim)  # output size: (B, 29, 512)
            x = x.transpose(1, 2)  # output size: (B, 512, 29)
            x = self.backend_conv1(x)  # output size: (B, 2048, 1)
            x = torch.mean(x, 2)  # output size: (B, 2048)
            x = self.backend_conv2(x)  # output size: (B, 500)
        elif self.mode == 'backendGRU' or self.mode == 'finetuneGRU':
            x = x.view(-1, self.frameLen, self.inputDim)  # output size: (B, 29, 512)
            x = self.gru(x)  # output size: (B, 29, 500)
        else:
            raise Exception('No model is selected')
        return x

```

"When you tried to use MFCC, did you put something between the MFCC feature and the BGRU layers ? Or did you feed the feature to the BGRU layer directly?"
No. We only use MFCC and feed the normalised feature to BGRU layer directly.

"Did you tried to truncate the LRW videos to isolate the words?"
Themos did a relative work:
@article{stafylakis2018pushing,
  title={Pushing the boundaries of audiovisual word recognition using Residual Networks and LSTMs},
  author={Stafylakis, Themos and Khan, Muhammad Haris and Tzimiropoulos, Georgios},
  journal={Computer Vision and Image Understanding},
  volume={176},
  pages={22--32},
  year={2018},
  publisher={Elsevier}

Hope this helps!

**Comment by baptistepouthier** [comment]
Thank you a lot, and congratulation for your work!
